AmplifyPlugins/API/AWSAPICategoryPlugin/Interceptor/SubscriptionInterceptor/AuthenticationTokenAuthInterceptor.swift

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AppSyncRealTimeClient
+import Amplify
+
+/// General purpose authenticatication subscriptions interceptor for providers whose only
+/// requirement is to provide an authentication token via the "Authorization" header
+class AuthenticationTokenAuthInterceptor: AuthInterceptor {
```

Comment:
This is largely based on `OIDCAuthInterceptor`.
All the AuthInterceptors (this one included) share a common logic, we should evaluate if it's feasible to refactor in a common interceptor with customization points that others provider-specific interceptors can use to inject custom logic.